app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -307,6 +307,10 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
             RustHttpConfig.setClient(lazy { components.core.client })
             RustLog.enable(components.analytics.crashReporter)
+            // We want to ensure Nimbus is initialized as early as possible so we can
+            // experiment on features close to startup.
+            // But we need viaduct (the RustHttp client) to be ready before we do.
+            components.analytics.experiments.initialize()
```

Comment:
From a performance perspective this is fine, as all the `initialize` code runs on a separate thread now (even if we didn't launch it on an IO scope here). The default `lazy` thread-safety mode also makes sure our lazy init block in `Analytics` will only run once even in case this code runs concurrently with another part of the app that also accesses `analytics.experiments`.

However, I am concerned about the `Nimbus` instance itself being accessed from multiple threads, in particular about the thread visibility of the mutable class fields. Thread visibility is not guaranteed for non final fields, IIRC, so it's possible that another thread sees `Nimbus.nimbus` and/or `Nimbus.datadir` as null. Some options are making methods on `Nimbus` synchronized or the affected fields `volatile`.

Reply:
Opened a PR in A-C: https://github.com/mozilla-mobile/android-components/pull/9618 (the `datadir` was already removed.)

Reply:
A-C change landed and upgraded here. So this is good now.